### PR TITLE
Add a warning message for the cases where an evaluation metric does n…

### PR DESCRIPTION
**Pull request by bigbird555**
…ot exist.

## Motivation

Please refer to the issue #1226 .

## Description of the changes

I added a warning message which will be displayed to users when there is no evaluation metric for pruning specified by them.

### CODE REVIEW

File: optuna/integration/keras.py [hunk 1]
```diff
@@ -57,6 +57,12 @@ def on_epoch_end(self, epoch: int, logs: Optional[Dict[str, float]] = None) -> N
         logs = logs or {}
         current_score = logs.get(self._monitor)
         if current_score is None:
+            message = (
+                "Warning: there is not an evaluation metric for pruning named "
+                + self._monitor
+                + " ."
+            )
+            print(message)
```
[the remark this answers is not in the document]
Comment (HideakiImamura) [reply optuna/integration/keras.py:65]: If we use `warnings.warn()`, the warning message can be simplified as follows: "There is not an evaluation metric for pruning named {}"

**Review by HideakiImamura — CHANGES_REQUESTED** [review]
Thanks for the PR! I totally agree with your idea to warn when the invalid metric is specified. The change looks basically good to me. I have some minor comments.

**Comment by bigbird555** [comment]
Thank you for your always thoughtful comments.
I completely understood the matter around coding conventions.
I'm going to follow your advice and fix the code.

File: optuna/integration/keras.py [hunk 2]
```diff
@@ -57,6 +57,10 @@ def on_epoch_end(self, epoch: int, logs: Optional[Dict[str, float]] = None) -> N
         logs = logs or {}
         current_score = logs.get(self._monitor)
         if current_score is None:
+            message = "There is not an evaluation metric for pruning named {}.".format(
+                self._monitor
+            )
+            warnings.warn(message)
```
Comment (toshihikoyanase) [inline optuna/integration/keras.py:63]: Please import `warnings` to use it.

@hvy  I think you expect an error here instead of `warnings` in https://github.com/optuna/optuna/issues/1226#issue-615516601. I guess raising the errors may break user code if the Keras returns `None` as expected values. But, I'm not sure if it occurs.
What do you think of it?

Comment (bigbird555) [reply optuna/integration/keras.py:65]: Thank you for your reply!
I just noticed this mistake and fixed it.

Comment (hvy) [reply optuna/integration/keras.py:65]: Thanks @toshihikoyanase for pinging. I think the current design to yield a warning sound good. You're right about the compatibility issue, and I also believe we have to be more careful if we are to raise errors, i.e. make sure that we understand the behavior of the integration framework.

Comment (toshihikoyanase) [reply optuna/integration/keras.py:65]: @hvy Thank you for your reply. That makes sense. I'll accept `warning`.

File: optuna/integration/keras.py [hunk 3]
```diff
@@ -1,6 +1,8 @@
 from typing import Dict
 from typing import Optional
 
+import warnings
+
```
Comment (toshihikoyanase) [inline optuna/integration/keras.py:5]: Please remove the blank line since `warnings` is a standard library.
```suggestion
import warnings

```

File: optuna/integration/keras.py [hunk 4]
```diff
@@ -57,6 +59,10 @@ def on_epoch_end(self, epoch: int, logs: Optional[Dict[str, float]] = None) -> N
         logs = logs or {}
         current_score = logs.get(self._monitor)
         if current_score is None:
+            message = "There is not an evaluation metric for pruning named {}.".format(
+                self._monitor
+            )
+            warnings.warn(message)
```
Comment (toshihikoyanase) [inline optuna/integration/keras.py:65]: Could you add a test case to check this warning to `tests/integration_tests/test_keras.py`?
We can use `with pytest.warns(UserWarning)` to assert the warning.

**Review by toshihikoyanase — CHANGES_REQUESTED** [review]
Thank you for your swift action. I added two comments.

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your update. I suggest an additional change. I'd like to hear your opinions about it.

File: optuna/integration/keras.py [hunk 5]
```diff
@@ -57,6 +58,10 @@ def on_epoch_end(self, epoch: int, logs: Optional[Dict[str, float]] = None) -> N
         logs = logs or {}
         current_score = logs.get(self._monitor)
         if current_score is None:
+            message = "There is not an evaluation metric for pruning named {}.".format(
```
[the remark this answers is not in the document]
Comment (bigbird555) [reply optuna/integration/keras.py]: I've changed the message to follow the feedback. It would be helpful if you could check it out.

File: optuna/integration/keras.py [hunk 6]
```diff
@@ -57,6 +58,12 @@ def on_epoch_end(self, epoch: int, logs: Optional[Dict[str, float]] = None) -> N
         logs = logs or {}
         current_score = logs.get(self._monitor)
         if current_score is None:
+            message = (
+                "The metric '{}' is not in the evaluation logs for pruning. "
+                "'{}' may not exist as evaluation metric for pruning. "
+                "Or '{}' may be a typo.".format(self._monitor, self._monitor, self._monitor)
```
Comment (toshihikoyanase) [inline optuna/integration/keras.py:64]: IMO, this message is a bit verbose because `self._monitor` appears three times in it. Can we simplify it, for example, as follows:

> "The metric '{}' is not in the evaluation logs for pruning. Please make sure you set the correct metric name."

Comment (bigbird555) [reply optuna/integration/keras.py:64]: If brevity is the priority, that seems better. I will revise the message that way.

**Review by HideakiImamura — APPROVED** [review]
Sorry for the late review. Thanks for the update! LGTM!

**Review by toshihikoyanase — APPROVED** [review]
I'm sorry for the delayed response. The change looks good to me. Thank you for your continuous contribution.

If you also apply similar changes to callbacks for tf.keras and PyTorch Lightning (#1226), it would be a great help.
